src/core/ngx_palloc.c

```diff
@@ -158,6 +162,15 @@ ngx_palloc_small(ngx_pool_t *pool, size_t size, ngx_uint_t align)
 
     p = pool->current;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+    /*
+     * We should be less than a page and thus precisely representable at
+     * any alignment.
+     */
+    assert(size == __builtin_cheri_round_representable_length(size));
+    assert(~__builtin_cheri_representable_alignment_mask(size) == 0);
+#endif
```

Comment:
CSetBoundsExact already asserts this?

Reply:
True, I guess I could just add a comment there.